repose-aggregator/components/filters/client-auth/src/test/groovy/org/openrepose/filters/clientauth/AuthNFilterDeprecationTest.groovy

```diff
@@ -0,0 +1,32 @@
+package org.openrepose.filters.clientauth
+
+import com.mockrunner.mock.web.MockFilterConfig
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.LoggerContext
+import org.apache.logging.log4j.test.appender.ListAppender
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.datastore.DatastoreService
+import spock.lang.Specification
+
+import static org.mockito.Mockito.mock
+
+class AuthNFilterDeprecationTest extends Specification {
+    def 'logs a deprecation warning upon init'() {
+        given:
+        System.setProperty("javax.xml.parsers.DocumentBuilderFactory",
```

Comment:
It is.

So yesterday we added an exclusion to the mockrunner dependency to try and win the classpath battle, but some of these tests were still failing. Actually, it seemed like the exclusion fixed some of the tests, and this System.setProperty jank fixed other tests. Neither one alone fixed all the tests.

Anyway, I don't think bringing the right XML parser onto the classpath alone fixes our issue. It has to be set to be used /somewhere/. That place may be in one of the libraries we pull in, but we shouldn't depend on it. This post does a decent job of explaining why:
https://www.databasesandlife.com/java-always-explicitly-specify-which-xml-parser-to-use/

Let me know if I've missed anything, or what you'd like to do from here. I'm going to default to leaving both fixes on the branch.